src/pages/home/ReportScreen.js

```diff
@@ -1,54 +1,108 @@
+    route: PropTypes.shape({
+        /* Route specific parameters used on this screen */
+        params: PropTypes.shape({
+            /* The ID of the report this screen should display */
+            reportID: PropTypes.string,
+        }).isRequired,
+    }).isRequired,
 };
 
-const ReportScreen = (props) => {
-    const activeReportID = parseInt(props.currentlyViewedReportID, 10);
-    if (!activeReportID) {
-        return null;
-    }
-
-    return (
-        <ScreenWrapper
-            style={[
-                styles.appContent,
-                styles.flex1,
-                styles.flexColumn,
-            ]}
-        >
-            <HeaderView
-                reportID={activeReportID}
-                onNavigationMenuButtonClicked={() => Navigation.navigate(ROUTES.HOME)}
-            />
-            <View style={[styles.dFlex, styles.flex1]}>
-                <ReportView reportID={activeReportID} />
-            </View>
-        </ScreenWrapper>
-    );
-};
+class ReportScreen extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isLoading: true,
+        };
+    }
+
+    componentDidMount() {
+        this.prepareTransition();
+    }
+
+    componentDidUpdate(prevProps) {
+        const reportChanged = this.props.route.params.reportID !== prevProps.route.params.reportID;
+
+        if (reportChanged) {
+            this.prepareTransition();
+        }
+    }
+
+    componentWillUnmount() {
+        clearTimeout(this.loadingTimerId);
+    }
+
+    /**
+     * Get the currently viewed report ID as number
+     *
+     * @returns {Number}
+     */
+    getReportID() {
+        const params = this.props.route.params;
+        return Number.parseInt(params.reportID, 10);
+    }
+
+    /**
+     * When reports change there's a brief time content is not ready to be displayed
+     *
+     * @returns {Boolean}
+     */
+    isReadyToDisplayReport() {
+        return !this.state.isLoading && Boolean(this.getReportID());
+    }
+
+    /**
+     * Configures a small loading transition of fixed time and proceeds with rendering available data
+     */
+    prepareTransition() {
+        if (!this.getReportID()) { return; }
+
+        this.setState({isLoading: true});
+
+        clearTimeout(this.loadingTimerId);
+        this.loadingTimerId = setTimeout(() => this.setState({isLoading: false}), 300);
+    }
+
+    render() {
+        return (
+            <ScreenWrapper
+                style={[
+                    styles.appContent,
+                    styles.flex1,
+                    styles.flexColumn,
+                ]}
+            >
+                <HeaderView
+                    reportID={this.getReportID()}
+                    onNavigationMenuButtonClicked={this.props.navigation.openDrawer}
+                />
+
+                <View style={[styles.dFlex, styles.flex1]}>
+                    <ReportView
+                        isReady={this.isReadyToDisplayReport()}
```

Comment:
Would it maybe be simpler to not pass `isReady` here and just have the full screen loader in this component?

Reply:
I think (but not sure) that we would want to hide the header as well if the report is still loading?

Reply:
Yeah good idea, I had it like that initially but tested different stuff by moving it around.
I noticed the Header information is there right away when you select a conversation and that's why I left it "outside the blanket" but maybe it adds to the overall experience so I'll cover it as well